File: search/models/booking.js
var mongoose = require('mongoose'),
    config = require('../helpers/env');

mongoose.Promise = require('bluebird');
mongoose.createConnection('mongodb://' + config.database() + '/restful-booker-platform');

var bookingSchema = mongoose.Schema({
    bookingid: {type: Number},
    firstname: { type: String, required: true},
    lastname: { type: String, required: true},
    totalprice: { type: Number, required: true},
    depositpaid: { type: Boolean, required: true},
    bookingdates: {
      checkin: { type: Date, required: true},
      checkout: { type: Date, required: true}
    },
    additionalneeds: { type: String, required: false}
  }, { versionKey: false });

var Booking = mongoose.model('Booking', bookingSchema);

exports.search = function(query, callback){
  Booking.find(query).select('-_id').exec(function(err, booking){
    if(err){
      callback(err);
    } else {
      callback(null, booking);
    }
  });
}

File: end-to-end-tests/src/main/java/models/Booking.java
package models;

public class Booking {

    private String firstname;
    private String lastname;
    private String totalPrice;

    public Booking(String firstname, String lastname, String totalPrice) {
        this.firstname = firstname;
        this.lastname = lastname;
        this.totalPrice = totalPrice;
    }

    public String getFirstname() {
        return firstname;
    }

    public String getLastname() {
        return lastname;
    }

    public String getTotalPrice() {
        return totalPrice;
    }
}
